source/fundamentals/auth.txt

```diff
@@ -272,102 +328,125 @@ using the following placeholders:
+with your relative URI:
 
-        If you prefer to pass the AWS session token in the connection string
-        alongside your ``MongoCredential``, specify your authentication mechanism
-        in the ``authMechanism`` parameter and your session token in the
-        ``authMechanismProperties`` parameter. Then, add it to your
-        ``MongoClientSettings`` by calling the
-        `applyConnectionString() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyConnectionString(com.mongodb.ConnectionString)>`__
-        method as follows:
+.. code-block:: bash
 
-        .. literalinclude:: /includes/fundamentals/code-snippets/MongoDbAwsAuth.java
-           :language: java
-           :dedent:
-           :start-after: start mongoCredentialSessionTokenConnString
-           :end-before: end mongoCredentialSessionTokenConnString
+   export AWS_CONTAINER_CREDENTIALS_RELATIVE_URI=<your ECS/EC2 endpoint>
 
-      - **Specify your AWS session token in a MongoCredential.**
+.. important::
 
-        You can include your AWS session token in your ``MongoCredential``
-        instance by specifying it in a call to the
-        `withMechanismProperty() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoCredential.html#withMechanismProperty(java.lang.String,T)>`__
-        method as shown below:
+   To authenticate using an ECS/EC2 endpoint, you must unset the following
+   environment variables because the driver otherwise attempts to authenticate
+   using them:
 
-        .. literalinclude:: /includes/fundamentals/code-snippets/MongoDbAwsAuth.java
-           :language: java
-           :dedent:
-           :start-after: start mongoCredentialSessionTokenCredential
-           :end-before: end mongoCredentialSessionTokenCredential
+   - ``AWS_ACCESS_KEY_ID``
+   - ``AWS_SECRET_ACCESS_KEY``
+   - ``AWS_SESSION_TOKEN``
 
-      - **Specify your AWS session token in an environment variable.**
+To learn more about the AWS container credential provider and container
+credentials, see the AWS documentation on `Container credentials <https://docs.aws.amazon.com/sdkref/latest/guide/feature-container-credentials.html>`__.
 
-        In your client execution environment, set an environment variable
-        called ``AWS_SESSION_TOKEN`` and assign your token to it. The value is
-        automatically picked up by your ``MongoClient`` when you specify the
-        ``MONGODB-AWS`` authentication mechanism.
+.. _java-mongodb-aws-refresh-credentials:
 
-Refresh Credentials
-+++++++++++++++++++
+Refresh your Credentials
+++++++++++++++++++++++++
 
-The driver supports refreshing credentials for cases such as assuming roles
-or using `Elastic Kubernetes Service <https://docs.aws.amazon.com/eks/latest/userguide/what-is-eks.html>`__.
+The driver supports refreshing credentials to support assuming roles
+or using the `Elastic Kubernetes Service <https://docs.aws.amazon.com/eks/latest/userguide/what-is-eks.html>`__.
```

Comment:
Issue: A lot of gerunds ("-ing" verbs) make this sentence a little confusing.

Suggestion: I'm not sure I have the technical knowledge to rephrase this accurately, unfortunately! Anything that reduces the number of gerunds should work.

Reply:
Thanks for spotting this. I didn't review the grammar of the existing content since that felt beyond the scope of what I was trying to accomplish, but could be difficult for the reviewer to identify. Will think of a better way to phrase it.